doc/user/migrating-existing-apis.md

```diff
@@ -0,0 +1,384 @@
+# Migrating Existing Kubernetes APIs
+
+Kubernetes APIs are assumed to evolve over time, hence the well-defined API [versioning scheme][k8s-versioning]. Upgrading your operator's APIs can be a non-trivial task, one that will involve changing quite a few source files and manifests. This document aims to identify the complexities of migrating an operator project's API using examples from existing operators.
+
+While examples in this guide follow particular types of API migrations, most of the documented migration steps can be generalized to all migration types. A thorough discussion of migration types for a particular project type (Go, Ansible, Helm) is found at the end of each project type's section.
+
```

Comment:
It may be beneficial to start with a happy case where the `GVK` contains a single kind.

Reply:
You mean do a walkthrough of a migration with a single kind?

Reply:
Correct - the first case is a bit more difficult to consume when compared to the case outlined [here](https://github.com/operator-framework/operator-sdk/pull/1680/files#diff-f417cc5a9376029258890e7aca5e5ccbR338).